lib/modules/manager/gomod/artifacts.ts

```diff
@@ -156,37 +165,7 @@ export async function updateArtifacts({
   const useVendor = (await readLocalFile(vendorModulesFileName)) !== null;
 
   try {
-    // Regex match inline replace directive, example:
-    // replace golang.org/x/net v1.2.3 => example.com/fork/net v1.4.5
-    // https://go.dev/ref/mod#go-mod-file-replace
-    const inlineReplaceRegEx = regEx(
-      /(\r?\n)(replace\s+[^\s]+\s+=>\s+\.\.\/.*)/g
-    );
-
-    // $1 will be matched with the (\r?n) group
-    // $2 will be matched with the inline replace match, example
-    // "// renovate-replace replace golang.org/x/net v1.2.3 => example.com/fork/net v1.4.5"
-    const inlineCommentOut = '$1// renovate-replace $2';
-
-    // Regex match replace directive block, example:
-    // replace (
-    //     golang.org/x/net v1.2.3 => example.com/fork/net v1.4.5
-    // )
-    const blockReplaceRegEx = regEx(/(\r?\n)replace\s*\([^)]+\s*\)/g);
-
-    /**
-     * replacerFunction for commenting out replace blocks
-     * @param match A string representing a golang replace directive block
-     * @returns A commented out block with // renovate-replace
-     */
-    const blockCommentOut = (match): string =>
-      match.replace(/(\r?\n)/g, '$1// renovate-replace ');
-
-    // Comment out golang replace directives
-    const massagedGoMod = newGoModContent
-      .replace(inlineReplaceRegEx, inlineCommentOut)
-      .replace(blockReplaceRegEx, blockCommentOut);
-
+    const massagedGoMod = disableLocalParentReplaceDirective(newGoModContent);
```

Comment:
This looks like a refactor and a fix PR, that should be done in separate PR for easier review.

Reply:
I was primarily interested in getting the bug fixed, but I didn't see an obvious way to test the replacement logic in isolation without splitting it into a separate function. If you have any resources on testing this using integration tests or on real repositories, I can revert the unit tests and test this PR that way.